vyper/parser/stmt.py

```diff
@@ -108,12 +108,18 @@ def _check_valid_assign(self, sub):
                 return
             else:
                 raise TypeMismatchException('Invalid type, expected: bytes32', self.stmt)
+        # This is dead code once anonymous structs are removed.
         elif isinstance(self.stmt.annotation, ast.Dict):
             if not isinstance(sub.typ, StructType):
                 raise TypeMismatchException('Invalid type, expected a struct')
         elif isinstance(self.stmt.annotation, ast.Subscript):
             if not isinstance(sub.typ, (ListType, ByteArrayType)):  # check list assign.
                 raise TypeMismatchException('Invalid type, expected: %s' % self.stmt.annotation.value.id, self.stmt)
+        elif isinstance(sub.typ, StructType):
+            # This needs to get more sophisticated in the presence of
+            # foreign structs.
```

Comment:
Off-topic: I am not sure I like the idea of foreign structs for clarity of reading a contract. :thinking: having all your struct info accessible in on place is really good way to read a contract, something to think about.

Reply:
That's a bridge we cross when we come to it lol.

Reply:
As part of the off-topic chain, I think the concepts in EIP712 may come in handy when we do get to this bridge. I like the idea of encoding the EIP712 domain directly into a contract definition for example.